style.md

```diff
@@ -1923,6 +1924,90 @@ sptr := &T{Name: "bar"}
 </td></tr>
 </tbody></table>
 
+### Initializing Maps
+
+Prefer `make(..)` for empty maps, and maps populated
```

Comment:
The reason I'd originally included a reference to the runtime hashmap implementation is because it's not obvious that size hints actually do not do anything unless the hint is large enough to require multiple buckets - which is different from slice allocation, which *does* allocate that memory based directly on the capacity.

I think it's fine to include a blanket recommendation to provide capacity, but it should probably also include a caveat that essentially says "just because you pass a hint doesn't necessarily mean that anything will be allocated".

What do you think?

Reply:
The optimizations for small size hints are interesting, but I think they end up being an implementation detail that the user shouldn't worry about -- if they have a size hint, they should pass it in.

The main difference between smaller/larger size hints seems to be:
 - For larger sizes, the `*hmap` and the buckets are pre-allocated.
 - For smaller maps, the `*hmap` is allocated, and the first assignment causes the bucket to be created.

In both cases, the total number of allocations is the same, but where the allocation happens can move around. It can definitely be a little surprising if a size hint was passed in and there still was an allocation on an assignment, but I don't think it's worth calling out specifically here.